src/Nethermind/Nethermind.Network/P2P/PacketSender.cs

```diff
@@ -49,19 +52,35 @@ public PacketSender(IMessageSerializationService messageSerializationService, IL
 
             IByteBuffer buffer = _messageSerializationService.ZeroSerialize(message);
             int length = buffer.ReadableBytes;
-            _context.WriteAndFlushAsync(buffer).ContinueWith(t =>
+
+            // Running in background
+#pragma warning disable CS4014
+            SendBuffer(buffer);
+#pragma warning restore CS4014
+
+            return length;
+        }
+
+        private async Task SendBuffer(IByteBuffer buffer)
+        {
+            try
             {
-                if (t.IsFaulted)
+                if (_sendLatency != TimeSpan.Zero)
                 {
-                    if (_context.Channel is { Active: false })
-                    {
-                        if (_logger.IsTrace) _logger.Trace($"Channel is not active - {t.Exception.Message}");
-                    }
-                    else if (_logger.IsError) _logger.Error("Channel is active", t.Exception);
+                    // Tried to implement this as a pipeline handler. Got a lot of peering issue for some reason...
+                    await Task.Delay(_sendLatency);
                 }
```

Comment:
I am tempted to suggest a decorator pattern here, but it might be an overkill/overdesign?

Reply:
I'd say yes in this case.